modal/secret.py

```diff
@@ -54,8 +55,40 @@ async def _load(resolver: Resolver, existing_object_id: Optional[str]) -> _Secre
             resp = await resolver.client.stub.SecretCreate(req)
             return _SecretHandle._from_id(resp.secret_id, resolver.client, None)
 
-        rep = f"Secret([{', '.join(env_dict.keys())}])"
-        super().__init__(_load, rep)
+        rep = f"Secret.from_dict([{', '.join(env_dict.keys())}])"
+        return _Secret._from_loader(_load, rep)
+
+    def __init__(self, env_dict: Dict[str, str]):
+        """`Secret({...})` is deprecated. Please use `Secret.from_dict({...})` instead."""
+        deprecation_warning(date(2023, 5, 1), self.__init__.__doc__)
+        obj = _Secret.from_dict(env_dict)
+        self._init_from_other(obj)
+
+    @staticmethod
+    def from_dotenv():
+        async def _load(resolver: Resolver, existing_object_id: Optional[str]) -> _SecretHandle:
+            try:
+                from dotenv import find_dotenv, dotenv_values
+            except ImportError:
+                raise ImportError(
+                    "Need the `dotenv` package installed. You can install it by running `pip install python-dotenv`."
+                )
+
+            # TODO(erikbern): dotenv tries to locate .env files based on the location of the file in the stack frame.
+            # Since the modal code "intermediates" this, a .env file in the user's local directory won't be picked up.
+            # We avoid this by just using the cwd instead.
+            dotenv_path = find_dotenv(usecwd=True)
```

Comment:
It's better than using the stack frame (since it doesn't break modal run) but it somehow still feels suboptimal and potentially confusing to use cwd instead of the directory of the run file. Not sure if we can easily get it to use that dir somehow...
One way would be to allow an optional path argument to `from_dotenv`, either with the env file's explicit path, or a path to search upwards from e.g. `Secret.from_dotenv(__file__)` for convenience?

Reply:
yeah good point, let me add that as an argument